src/controllers/HistoryController.ts

```diff
@@ -153,9 +195,79 @@ export class HistoryController extends RootComponent {
     return value;
   }
 
-  public debugInfo() {
-    return {
-      state: this.model.getAttributes()
-    };
+  private mapStateDifferenceToUsageAnalyticsCall(stateDifference: IStringMap<any>) {
```

Comment:
In those case, we really do want to log only one event, and the most important one first. It's also a "best" effort, as in it's not really recommended to modify the query by modifying the URL with custom code, but we will still try to support it if it can remove some complexity for users of our stuff.

There's no way to log multiple search cause for a single search event, either in the UI, or the backend service (UA).

I can add a comment that explain the hierarchy of event in order of importance ( where q is arguably the most important one.)